### PR TITLE
fix: Fixes various issues with compiling through execution nodes

### PR DESCRIPTION
**Pull request by tssweeney**
Fixes:
* https://wandb.atlassian.net/browse/WB-14101
* https://wandb.atlassian.net/browse/WB-14124

Run-history plots working:
Screenshot 2023-06-26 at 22 24 09

Ok, so i realize that most of these issues only come up with TimeSeries panel, but i wanted to fix it because these patterns with lazy nodes are becoming more popular. Essentially the user experience was that feeding an input node to TimeSeries panel causes crashes. There are two main issues

1. Since we are lazily executing, we get `execute` ops sprinkled into the graph. Stitch was not flowing through that
2. Since we are lazily executing, we get `function-__call__` ops sprinkled into the graph. We now safely compile these away
3. Once you fix the above issues, we realize that compile_domain does not properly merge `parquetHistory`. We now solve that.

Other Notes:
* Fixed tslint issue
* compile_domain more gracefully handles internal errors
* panel_time_series correctly executes it's inner functions
* Fixed PanelTimeSeries (and related ops) to work with smaller time frames
* `define_fn` now returns a runtime node

### CODE REVIEW

File: weave/ecosystem/wandb/panel_time_series.py [hunk 1]
```diff
@@ -197,35 +197,54 @@ def initialize(self) -> TimeSeriesConfig:
...
+            label_node.val,
+            label_node.type,
+        )
+
+        agg_node = weave_internal.define_fn(
+            {"group": input_node.type},  # type: ignore
+            lambda group: group.count(),
+        )
+
+        agg = weave_internal.const(
```
Comment (shawnlewis) [inline weave/ecosystem/wandb/panel_time_series.py:239]: Why did you need to wrap in another const node here?

Did you check if you can still configure panel_time_series from the UI?

Comment (tssweeney) [reply weave/ecosystem/wandb/panel_time_series.py:239]: Oh - so we need all these to be RuntimeConstNodes, not just ConstNodes. This might just be an artifact of testing flow, but doing something like this (which mimics `initialize`):
```
    init_config_node = panel.initialize()
    init_config = weave.use(init_config_node)
    panel.config = init_config
    render_node = panel.render()
    res = weave.use(render_node)
```

The `render` op gets called with these nodes and internally expects Runtime nodes.

If this is a pattern violation, happy to change

Comment (shawnlewis) [reply weave/ecosystem/wandb/panel_time_series.py:239]: I totally missed that you were just rewrapping these, I thought you were double wrapping...

Well, the one thing about these plots is I was trying to have them read really well, like be implemented as we would want a user to do.

Having to change nodes to dispatch nodes is no good, also using better_call_fn instead of the __call__ dunder.

Comment (tssweeney) [reply weave/ecosystem/wandb/panel_time_series.py]: fixed (: Made define_fn return runtime node
